bundles/org.openhab.core/src/test/java/org/openhab/core/util/StringUtilsTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.core.util;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.Test;
+
+/**
+ * The {@link StringUtils} class defines some static string utility methods
+ *
+ * @author Leo Siepel - Initial contribution
+ */
+@NonNullByDefault
+public class StringUtilsTest {
+
```

Comment:
Missing test for ```getRandomString```. It could  be worth the effort to check if every character of the ```charset``` is picked roughly at the same probability. Implementation/rounding issues could lead to the problem that the last character in the set is not picked at the same probability as others. Maybe a short test that getRandomString(10000, "123") contains >30% of each character.
Consider this just as a suggestion.

Reply:
Test added